src/classes/CRLP_RollupProcessor.cls

```diff
@@ -0,0 +1,538 @@
+     * Note that this method completely clones the rollup definitions so that the updated references
+     * do not overwrite the original references, which can create issues when there are multiple instances
+     * of these collections for multiple parent records.
+     */
+    public CRLP_RollupProcessor withRollupDefinitions(Map<Id, List<CRLP_Rollup>> rollupDefsByParentId) {
+        this.rollupsDefsByParentId = new Map<Id, List<CRLP_Rollup>>();
+        for (Id parentId : rollupDefsByParentId.keySet()) {
+            if (rollupDefsByParentId.get(parentId) != null) {
+                List<CRLP_Rollup> clonedList = CRLP_RollupProcessor_SVC.fullyCloneList(rollupDefsByParentId.get(parentId));
+                this.rollupsDefsByParentId.put(parentId, clonedList);
+            }
+        }
+        return this;
+    }
+
+    /**
+     * @description Inject a previously defined set of RollupIds for a single Parent Id. Optional
+     */
+    public CRLP_RollupProcessor withRollupDefinitions(List<CRLP_Rollup> rollupDefs) {
+        return withRollupDefinitions(new Map<Id, List<CRLP_Rollup>>{
+                parentRecords[0].Id => rollupDefs
+        });
+    }
+
+
+    /**
+     * @description Inject a map of previously processed Opportunity Id's by Parent Id. Optional
+     * This is needed in SkewMode to support rollups that span multiple batch iterations.
+     */
+    public CRLP_RollupProcessor withPreviouslyProcessedOppIdsMap(Map<Id,Set<Id>> processedOppIdsByParentId) {
+        previouslyProcessedOppIdsByParentId = processedOppIdsByParentId;
+        return this;
+    }
+
+    /**
+     * @description Determine the SObjectType of the Detail Records based on the current rollup type
+     */
+    private SObjectType detailObjectType {
+        get {
+            if (detailObjectType == null) {
+                detailObjectType = CRLP_Rollup_SVC.getDetailObjectType(rollupType);
+            }
+            return detailObjectType;
+        } set;
+    }
+
+    /**
+     * @description Determine the SObjectType of the Detail Records based on the current rollup type
+     */
+    private String parentRelationshipField {
+        get {
+            if (parentRelationshipField == null) {
+                parentRelationshipField = CRLP_Rollup_SVC.getParentRelationshipFieldForProcessingJob(rollupType);
+            }
+            return parentRelationshipField;
+        } set;
+    }
+
+    // ================================================================================================
+
+    /**
+    * @description Returns the Parent Object Field by parsing the parentKeyFieldFull property
+    */
+    private String parentObjectField {
+        get {
+            if (parentObjectField == null) {
+                // The parentKeyField field may be have dot-notation (ex: Contact__r.AccountId)
+                // If this is the case, need to parse out the two components to use
+                String keyField = parentRelationshipField;
```

Comment:
No need for `keyField` var, can use `parentRelationshipField`.

If the `parentRelationshipField` is just `AccountId` or `Account__c`, then `parentObjectField` will always be null. Thus, it is best to set this field to empty string by default, and whenever the value is checked (line 152), use `String.isBlank(parentObjectField)` and not use `parentObjectField == null`. This is applicable for all strings, let's use String methods.

Reply:
What would you think about leaving the `get` format, but changing the checks for `null` into `String.isNotBlank(parentRelationshipField)`?

Reply:
Yes, the `get` format stays for the `parentRelationshipField `, and we use `String` methods when checking String throughout (for this and other String properties).